sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -39,7 +40,10 @@ case class SortMergeJoinExec(
     joinType: JoinType,
     condition: Option[Expression],
     left: SparkPlan,
-    right: SparkPlan) extends BinaryExecNode with CodegenSupport {
+    right: SparkPlan,
+    requiredNumPartitions: Option[Int] = None,
+    hashingFunctionClass: Class[_ <: HashExpression[Int]] = classOf[Murmur3Hash])
```

Comment:
I think this can be done in a followup. For the first version we can just add a `HiveHashPartitioning`, which can satisfy `ClusteredDistribution`(save shuffle for aggregate) but not `HashClusteredDistribution`(can't save shuffle for join).